packages/widget_toolkit_pin/lib/src/lib_pin_code_with_biometrics/blocs/pin_code_bloc.dart

```diff
@@ -164,18 +164,35 @@ class PinCodeBloc extends $PinCodeBloc {
 
   /// Authenticates the user with biometrics after which the pin code is
   /// retrieved from the device and checked.
-  Future<bool?> _authenticateWithBiometrics() async {
+  Future<dynamic> _authenticateWithBiometrics() async {
     if (!await biometricAuthenticationService.isDeviceSupported) {
       throw ErrorEnableBiometrics(BiometricsMessage.notSupported);
     }
     if (await biometricAuthenticationService.authenticate(localizedReason)) {
       final pinCode = await pinCodeService.getPinCode();
       if (pinCode != null) {
-        return await _checkPin(pinCode, pinCode.length);
+        return _validateBiometricsPin(pinCode);
       }
     }
     return false;
   }
 
+  /// Checks the validity of the pin code retrieved from local storage
+  /// after biometric authentication is successful
+  Future<dynamic> _validateBiometricsPin(String pinCode) async {
```

Comment:
I think having two separate methods for handling pin validation is fine since the pin code that is being validated in the biometrics is already encrypted and it shouldn't be checked against the one that is typed in the keyboard.
Having a single unified method increases the complexity and makes code harder to understand

Reply:
I believe here we have a bigger problem. The pin codes should not be compared in the mobile application at all. The app should send the encrypted pin to the API so that it can be decrypted and verified from the BE side.

Reply:
Yes, you are right, it's unnecessary.
I have removed that method entirely